plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ConfigReader } from '@backstage/config';
+import { JsonObject } from '@backstage/types';
+import { createSentryCreateProjectAction } from './createProject';
+import { ActionContext } from '@backstage/plugin-scaffolder-node';
+import { InputError } from '@backstage/errors';
+import { randomBytes } from 'crypto';
+
+describe('sentry:project:create action', () => {
+  const createScaffolderConfig = (configData: JsonObject = {}) => ({
+    config: new ConfigReader({
+      scaffolder: {
+        ...configData,
+      },
+    }),
+  });
+
+  let fetch: jest.Func;
+
+  const mockFetch = (response = {}) => {
+    const mockedResponse = {
+      status: 201,
+      headers: {
+        get: () => 'application/json',
+      },
+      json: async () =>
+        Promise.resolve({
+          detail: 'project creation mocked result',
+        }),
+      text: async () => Promise.resolve('Unexpected error.'),
+      ...response,
+    };
+
+    fetch = jest.fn().mockImplementation(() => Promise.resolve(mockedResponse));
+    return mockedResponse;
+  };
+
+  const getActionContext = (): ActionContext<{
+    organizationSlug: string;
+    teamSlug: string;
+    name: string;
+    slug?: string;
+    authToken?: string;
+  }> => ({
+    workspacePath: './dev/proj',
+    createTemporaryDirectory: jest.fn(),
+    logger: jest.createMockFromModule('winston'),
+    logStream: jest.createMockFromModule('stream'),
+    input: {
+      organizationSlug: 'org',
+      teamSlug: 'team',
+      name: 'test project',
+      authToken: randomBytes(5).toString('hex'),
+    },
+    output: jest.fn(),
+  });
+
+  beforeEach(() => {
+    mockFetch();
+  });
+
+  test('should request sentry project create with specified parameters.', async () => {
+    const action = createSentryCreateProjectAction(createScaffolderConfig(), {
+      fetch,
+    });
+
+    const actionContext = getActionContext();
+
+    await action.handler(actionContext);
+
+    expect(fetch).toHaveBeenNthCalledWith(
```

Comment:
When switching to msw, you can put expects directly _inside the request handler_ and then putting an `expect.assertions(2)` or whatever in the test to assert that that assertion (1) really does get exercised as part of the test.